### DISCUSSION TITLE
Align face before swap, dynamic mask padding

### DISCUSSION THREAD

**Discussion by mlstudent2** [Bing-su/adetailer#343] [Ideas]
Great extension. I find myself editing a few values per image that I would love to see built in.

1. I've noticed that inpainting produces much better face swaps when the face is aligned upright. Is it possible to add an option to align a face before a swap, perform the swap on the aligned face, then "unrotate" the swapped face back to the original alignment before compositing?

2. Mask padding in absolute pixel values is kind of frustrating because the best value depends on the size of the face within the image. A close up requires more padding, a far away face requires less padding. It would be great if there was a way to set mask padding based on a percent of face size so that one value would work in most cases.

**Comment by Bing-su** [comment]
1. 이 기능은 간단한게 아닙니다. [ototadana/sd-face-editor](https://github.com/ototadana/sd-face-editor) 같은 확장을 사용해주세요.
2. 괜찮은 생각입니다. 그런데 패딩의 구현은 여기가 아니라 automatic1111에 있습니다.

**Reply by mlstudent2** [reply]
Thanks for the link, I have never seen that extension. Unfortunately I get much worse results vs Adetailer, maybe I just need to find the right settings.
